pkg/controller/dnszone/dnszone_controller_test.go

```diff
@@ -973,16 +1040,25 @@ func TestSetConditionsForErrorForAWS(t *testing.T) {
 			zr.SetConditionsForError(tc.error)
 
 			// Assert
-			if tc.expectCondition != nil {
-				for _, cond := range zr.dnsZone.Status.Conditions {
-					assert.Equal(t, cond.Type, tc.expectCondition.Type)
-					assert.Equal(t, cond.Status, tc.expectCondition.Status)
-					assert.Equal(t, cond.Reason, tc.expectCondition.Reason)
-					assert.Equal(t, cond.Message, tc.expectCondition.Message)
+			if tc.expectConditions != nil {
+				for _, expectedCondition := range tc.expectConditions {
+					foundCondition := false
+					for _, cond := range zr.dnsZone.Status.Conditions {
+						if expectedCondition.Type == cond.Type {
+							foundCondition = true
+							assert.Equal(t, cond.Status, expectedCondition.Status)
+							assert.Equal(t, cond.Reason, expectedCondition.Reason)
+							assert.Equal(t, cond.Message, expectedCondition.Message)
+						}
+
+					}
+					assert.True(t, foundCondition, "Did not find expected condition of type %s", expectedCondition.Type)
```

Comment:
Can you change it to [this](https://github.com/openshift/hive/blob/master/pkg/test/assert/assertions.go#L68-L77) pattern? We don't want to ever be reliant on the sequence of conditions when we check for them since conditions can be applied in any order